### ISSUE TITLE
{Worker}[Sprite]: Can't find sprite named 'dot.sdf'

### ISSUE THREAD

**Issue by friedbunny** [mapbox/mapbox-gl-native#1877]
Console spew when using the satellite style. Other styles have different spew — think I saw failed `marsh-16` loads on Streets. Will post when I see it again.

```
17:49:13.494 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
17:49:13.496 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
17:49:13.499 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
17:49:13.500 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
17:49:13.509 Mapbox GL [ERROR] {Map}[OpenGL]: stencil mask overflow
17:49:13.560 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
17:49:13.572 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
17:49:13.573 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
17:49:13.574 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
17:49:13.575 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
17:49:13.798 Mapbox GL [ERROR] {Map}[OpenGL]: stencil mask overflow
17:49:25.812 Mapbox GL [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
```

/cc @tmpsantos @incanus @kkaefer @ansis

**Comment by 1ec5** [comment]
This is because the Satellite style lacks a sprite sheet. The workaround for now is to have `-mapView:imageForAnnotation:` return an image when the Satellite style is active.

**Comment by picciano** [comment]
I can confirm the "marsh-16" spewing when using Streets...

```
2015-07-13 09:54:10.837 TallyGo[38640:15692583] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-13 09:54:10.837 TallyGo[38640:15692583] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-13 09:54:10.837 TallyGo[38640:15692583] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-13 09:54:10.837 TallyGo[38640:15692583] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
```

**Comment by kkaefer** [comment]
This is an intentional message informing you that the stylesheet/data references an image that is not available in the sprite.

**Comment by 1ec5** [comment]
@picciano, this is distinct from the issue @friedbunny reports above. It’s surprising, too, because a “marsh-16” sprite is defined in the [Mapbox Streets sprite sheet](https://github.com/mapbox/mapbox-gl-styles/blob/ios-b1/sprites/mapbox-streets.json#L1890).

**Comment by picciano** [comment]
I built from source (master) this morning. This is seen using the default style.

Does https://www.mapbox.com/mapbox-gl-styles/sprites/mapbox-streets.json somehow get cached on the client?

**Comment by picciano** [comment]
I removed the app and reinstalled. The warning went away. Something must be getting cached somewhere.

**Comment by picciano** [comment]
Warning came back after a couple more launches. Something is definitely a bit off.

**Comment by tmpsantos** [comment]
@picciano I think I know what is going on. The sprites are loaded asynchronously, so you get something on the screen and later when the sprites arrive we trigger an update and everything is rendered normally (and “marsh-16” is found).

Maybe we should print the warning only after the sprites are loaded/parsed, otherwise the warning is misleading.

**Comment by friedbunny** [comment]
`marsh-16` problems → #1885

**Comment by picciano** [comment]
That makes sense...i am seeing it with other sprites as well now. It does appear to be timing related.

```
2015-07-13 13:53:05.066 TallyGo[2648:906379] [INFO] {Worker}[Sprite]: Can't find sprite named 'camera-11'
2015-07-13 13:53:05.081 TallyGo[2648:906383] [INFO] {Worker}[Sprite]: Can't find sprite named 'camera-11'
2015-07-13 13:53:12.530 TallyGo[2648:906381] [INFO] {Worker}[Sprite]: Can't find sprite named 'camera-11'
2015-07-13 13:53:12.531 TallyGo[2648:906381] [INFO] {Worker}[Sprite]: Can't find sprite named 'camera-11'
```

**Comment by 1ec5** [comment]
It looks like all these messages are related. In https://github.com/mapbox/mapbox-gl-native/issues/1877#issuecomment-120540336, I suggested that Satellite was missing a sprite sheet, but that causes a different message, and only when attempting to display a pin annotation on the map. Let’s track the console output here in #1885.

**Comment by vinod1988** [comment]
i have this message on log console:

[INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:09.938 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:09.938 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:09.939 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:12.731 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:12.731 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:12.731 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:12.731 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:12.871 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:12.871 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:12.871 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:12.872 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:13.007 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:13.007 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:13.007 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:13.008 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:13.095 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:13.095 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:13.096 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'
2015-07-22 20:43:13.096 SwiftMenu[16266:206335] [INFO] {Map}[Sprite]: Can't find sprite named 'marsh-16'

**Comment by 1ec5** [comment]
@vinod1988, those console messages are harmless. They should be fixed by #1885 in the next release.

**Comment by iceman201** [comment]
I got same problem as [INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'
Do you guys know how to fix it?

**Comment by 1ec5** [comment]
@iceman201, what version of the iOS SDK are you using, and what style? In the past, the Satellite style (specifically, [satellite-v7](https://github.com/mapbox/mapbox-gl-styles/blob/master/styles/satellite-v8.json)) has been known to trigger this console output. However, in recent versions of the iOS SDK, satellite-v8 no longer refers to dot.sdf.

**Comment by kkaefer** [comment]
@iceman201 these messages are informational (hence `INFO`) telling you that you might be referencing a sprite that can't be found; the map should be displayed regardless of these messages.

**Comment by iceman201** [comment]
@1ec5 Mapbox-iOS-SDK (3.0.0) is the version I using now

I setup up as mapView = MGLMapView(frame: view.bounds, styleURL: MGLStyle.hybridStyleURL())
just this one

**Comment by iceman201** [comment]
@1ec5 it's mapbox://styles/mapbox/satellite-hybrid-v8

**Comment by 1ec5** [comment]
Ah, the Hybrid style does [refer to dot.sdf](https://github.com/mapbox/mapbox-gl-styles/blob/68deb34b1daa34dfdb0d22b452b00abf08185848/styles/satellite-hybrid-v8.json#L1262).

@kkaefer, although that console message is harmless, it’d be nice to eliminate any unnecessary console output. Does the style need to be modified, or gl-native?

**Comment by iceman201** [comment]
how can I refer a sprite by swift coding?
[INFO] {Worker}[Sprite]: Can't find sprite named 'dot.sdf'

**Comment by 1ec5** [comment]
@iceman201, the reference to dot.sdf can’t be accessed from Swift (or Objective-C); it’s baked into the style you’re using. It may be possible to modify the Hybrid style to stop referring to this sprite, using Mapbox Studio.

**Comment by kkaefer** [comment]
The style needs to be modified, by changing the name of the referenced image to one that actually exists, or by adding the image.

**Comment by iceman201** [comment]
@kkaefer mapView = MGLMapView(frame: view.bounds, styleURL: MGLStyle.hybridStyleURL()) 
in this sentence? or how?

**Comment by 1ec5** [comment]
It looks like the style in the mapbox/mapbox-gl-styles project needs to be corrected. For now, the easiest fix is to design your own custom style:

1. In [Mapbox Studio](https://www.mapbox.com/studio), create a new style based on the “Satellite Streets” template.
1. Judging from where “dot.sdf” occurs in the [style’s JSON code](https://github.com/mapbox/mapbox-gl-styles/blob/68deb34b1daa34dfdb0d22b452b00abf08185848/styles/satellite-hybrid-v8.json#L1262), select the “place_label_city_medium” layer under the “Place labels” folder.
1. Switch to the Icon tab.
1. You should see a warning on the Image field. Clear that field.
1. Hit Publish, then copy the style URL.
1. Back inside your view controller, change that line of code to:

```swift
mapView = MGLMapView(frame: view.bounds, styleURL: NSURL(string: "mapbox://styles/iceman201/asdfghjkl"))
# …or whatever the style URL happens to be
```

**Comment by andrewcar** [comment]
I've deleted all of the missing sprites such as "marsh-16," "dot-9," "dot-10," "circle-12," "airport-11" and "us-interstate-3-small" in various Icon tab -> Image fields or Pattern fields indicated by the warnings in orange. I still get a bunch of logs after publishing and building now for some new missing sprites I've never heard of such as "park-11," "us-interstate-2-small," "rail-11," "cemetery-11," "college-11," and a couple of old favorites like "airport-11" and "us-interstate-3-small." 

I don't know where these missing sprites are being referenced in Mapbox Studio but I have gone through every tab. I tried waiting a bit to see if the Publish hadn't kicked in but a few minutes later and I still get several logs. I can tell that it looking for these missing files is lowering performance and responsiveness in the map because as soon as I stop adding the annotation to the view, the map runs smoothly.
